Build error C1083 when PKG_CONFIG_PATH contains spaces [bug]

### Steps to reproduce

1.  Build and install the C library `libversion` to some directory with a space character in the path. E.g to `C:/Program Files/libversion` on Windows:

		set LIBVERSION_REF=3.0.1
		choco install -y git cmake visualstudio2019buildtools pkgconfiglite
		git clone https://github.com/repology/libversion.git --branch="%LIBVERSION_REF%"
		cd libversion
		cmake . -G "Visual Studio 15 2017 Win64"
		cmake --build . --config Release
		cmake --install .
		setx PKG_CONFIG_PATH "C:/Program Files/libversion/lib/pkgconfig"

2.  Check that it is discoverable by `pkg-config`:

		pkg-config --libs --cflags libversion

2.  Try to build the Python module `libversion`:

		git clone https://github.com/repology/py-libversion.git
		cd py-libversion
		python setup.py build

#### Actual result
~~~
(venv) D:\GIT\thirdparty-tracking\py-libversion>pkg-config --libs --cflags libversion
-IC:/Program\ Files/libversion/include -LC:/Program\ Files/libversion/lib -lversion

(venv) D:\GIT\thirdparty-tracking\py-libversion>python setup.py build
running build
running build_py
creating build
creating build\lib.win-amd64-3.8
creating build\lib.win-amd64-3.8\libversion
copying libversion\__init__.py -> build\lib.win-amd64-3.8\libversion
copying libversion\py.typed -> build\lib.win-amd64-3.8\libversion
copying libversion\_libversion.pyi -> build\lib.win-amd64-3.8\libversion
running build_ext
building 'libversion._libversion' extension
creating build\temp.win-amd64-3.8
creating build\temp.win-amd64-3.8\Release
creating build\temp.win-amd64-3.8\Release\src
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:/Program\ -ID:\GIT\thirdparty-tracking\py-libversion\venv\
include -IC:\Programs\Python38\include -IC:\Programs\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\include" "-IC:\Program Files (x86)\Micros
oft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" "-IC:\
Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" "-IC:\P
rogram Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" /Tcsrc/_libversion.c /Fobuild\temp.win-amd64-3.8\Release\src/_libversion.obj
_libversion.c
src/_libversion.c(24): fatal error C1083: Не удается открыть файл включение: libversion/version.h: No such file or directory,
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
~~~

#### Expected result
No error C1083.


### Root cause

From the manual [pkg-config(1)](http://manpagez.com/man/1/pkg-config/) ([sources](https://gitlab.freedesktop.org/pkg-config/pkg-config/-/blob/d97db4fae4c1cd099b506970b285dc2afd818ea2/pkg-config.1#L30-35)):

> Here is a typical usage scenario in a Makefile:
> 	
>     program: program.c
>     	cc program.c `pkg-config --cflags --libs gnomeui`

According to it, pkg-config outputs a shell-like string. It is wrong to split such string into command line arguments using any whitespace characters as the delimiter.

--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,6 @@
 #!/usr/bin/env python3
 
+import shlex
 import subprocess
 import sys
 from os import path
@@ -12,7 +13,8 @@ here = path.abspath(path.dirname(__file__))
 
 def pkgconfig(package):
     result = {}
-    for token in subprocess.check_output(['pkg-config', '--libs', '--cflags', package]).decode('utf-8').split():
+    pkg_config_output = subprocess.check_output(['pkg-config', '--libs', '--cflags', package]).decode('utf-8')
+    for token in shlex.split(pkg_config_output):
         if token.startswith('-I'):
             result.setdefault('include_dirs', []).append(token[2:])
         elif token.startswith('-L'):

